wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -1622,14 +1627,21 @@ protected JSch getJSch(final OpenSshConfig.Host hc, FS fs) throws JSchException
                     String password = remoteUrl.substring(remoteUrl.lastIndexOf(":") + 1, remoteUrl.indexOf("@"));
                     command.setCredentialsProvider(new UsernamePasswordCredentialsProvider(username, password));
                 } else {
+                    String gitUser = request.getAttributes().get("userName");
+                    String gitPassword = request.getAttributes().get("password");
+                    if (gitUser != null && gitPassword != null) {
+                        useRequestCredentials = true;
+                    }
+                    if (useRequestCredentials) {
+                        GitBasicAuthenticationCredentialsProvider.setCurrentCredentials(gitUser, gitPassword);
+                    }
                     credentials = credentialsLoader.getUserCredential(remoteUrl);
                     if (credentials != null) {
                         command.setCredentialsProvider(new UsernamePasswordCredentialsProvider(credentials.getUserName(),
-                                                                                               credentials.getPassword()));
```

Comment:
this was formatted according to https://github.com/codenvy/codenvy-dev-resources/wiki/Formatting#for-assignment-operator-the-break-comes-after-the-symbol